youtube_dl/extractor/ciscolive.py

```diff
@@ -0,0 +1,143 @@
+                'ext': 'mp4',
+                'title': '13 Smart Automations to Monitor Your Cisco IOS Network [BRKNMS-2465]',
+                'description': 'md5:171c3a1c0469c126d01f083a83d6c60b',
+                'timestamp': 1530305395,
+                'uploader_id': '5647924234001',
+                'upload_date': '20180629',
+                'location': '16B Mezz.',
+            },
+        },
+        {
+            'url': 'https://ciscolive.cisco.com/on-demand-library/?search.event=ciscoliveus2018&search.technicallevel=scpsSkillLevel_aintroductory&search.focus=scpsSessionFocus_designAndDeployment#/',
+            'md5': '993d4cf051f6174059328b1dce8e94bd',
+            'info_dict': {
+                'upload_date': '20180629',
+                'title': 'DevNet Panel-Applying Design Thinking to Building Products in Cisco [DEVNET-1794]',
+                'timestamp': 1530316421,
+                'uploader_id': '5647924234001',
+                'id': '5803751616001',
+                'description': 'md5:291dbd447bf745d1f61d944d9508538f',
+                'location': 'WoS, DevNet Theater',
+                'ext': 'mp4',
+            },
+        },
+        {
+            'url': 'https://ciscolive.cisco.com/on-demand-library/?search.technology=scpsTechnology_applicationDevelopment&search.technology=scpsTechnology_ipv6&search.focus=scpsSessionFocus_troubleshootingTroubleshooting#/',
+            'md5': '80e0c3b87e373fe3a3316b934b8915bf',
+            'info_dict': {
+                'upload_date': '20180629',
+                'title': 'Beating the CCIE Routing & Switching [BRKCCIE-9162]',
+                'timestamp': 1530311842,
+                'uploader_id': '5647924234001',
+                'id': '5803735679001',
+                'description': 'md5:18bf6e8a634df0a51290401f209089b0',
+                'location': 'Tulúm 02',
+                'ext': 'mp4',
+            },
+        }
+    ]
+
+    # These appear to be constant across all Cisco Live presentations
+    # and are not tied to any user session or event
+    RAINFOCUS_API_URL = 'https://events.rainfocus.com/api/%s'
+    RAINFOCUS_APIPROFILEID = 'Na3vqYdAlJFSxhYTYQGuMbpafMqftalz'
+    RAINFOCUS_WIDGETID = 'n6l4Lo05R8fiy3RpUBm447dZN8uNWoye'
+    BRIGHTCOVE_URL_TEMPLATE = 'http://players.brightcove.net/5647924234001/SyK2FdqjM_default/index.html?videoId=%s'
+
+    def _parse_rf_item(self, rf_item):
+        ''' Parses metadata and passes to Brightcove extractor
+
+        '''
+        event_name = rf_item.get('eventName')
+        cl_id = rf_item.get('abbreviation')
+        title = rf_item.get('title')
+        description = clean_html(rf_item.get('abstract'))
+        presenter_name = try_get(rf_item, lambda x: x['participants'][0]['fullName'])
+        presenter_title = try_get(rf_item, lambda x: x['participants'][0]['jobTitle'])
+        pdf_url = try_get(rf_item, lambda x: x['files'][0]['url'])
+        bc_id = try_get(rf_item, lambda x: x['videos'][0]['url'])
```

Comment:
Mandatory.

Reply:
By the time `_parse_rf_item()` is called, we have already verified that `bc_id` exists as an int with `_check_bc_id_exists()`. I don't think it makes sense to throw an exception, because several search query URLs might include one or two presentations that don't have video URLs, but we want to silently continue with the ones *with* video URLs. Or have I misunderstood?

Reply:
If you guarantee something exists you don't need to use `try_get`. The purpose of `try_get` is to be used when there are no such guarantee.